src/workloads/execution/FilterWithComplexLogicalExpression.yml

```diff
@@ -32,7 +32,108 @@ GlobalDefaults:
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter:
+                {$expr:
+                   {$or:
+                      {^Array: {of: {$and: [
+                        {$eq:
+                           ["$a",
+                            {^RandomInt:
+                               {min: 2, max: { ^Parameter: { Name: "aMaxValue", Default: 1000}}}}]},
+                        {$eq:
+                           ["$b",
+                            {^RandomInt:
+                               {min: 2, max: {^Parameter: { Name: "bMaxValue", Default: 1000}}}}]}]},
+                                number: {^Parameter: { Name: "NumberOfClauses", Default: 50}}}}}}
+
+- TemplateName: MatchOrWithManyChildClausesTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "MatchOr"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter:
+                { $or:
+                    { ^Array:
+                        { of:
+                            { a: { ^RandomInt:
+                                     { min: 2, max: { ^Parameter: { Name: "aMaxValue", Default: 1000 } } } },
+                              b: { ^RandomInt:
+                                     { min: 2, max: { ^Parameter: { Name: "bMaxValue", Default: 1000 } } } } },
+                          number: { ^Parameter: { Name: "NumberOfClauses", Default: 50 } } } } }
+
+- TemplateName: MatchAndWithManyChildClausesTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "MatchAnd"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {^Parameter: {Name: "Filter", Default: {}}}
+
+Array: &Array
+  {^Cycle: {ofLength: 1, fromGenerator: {^Array: {number: 200, of: {^Inc: {start: 0, step: 1}}}}}}
+
+FiftyClauseRootedAnd: &FiftyClauseRootedAnd
+  {$and:
+     {^Cycle: {ofLength: 1, fromGenerator:
```

Comment:
Ditto comment about why we need `^Cycle`.

Reply:
We have another open thread above this topic, so I'll resolve this one. I do think adding comments to explain the use of `^Cycle` makes sense.